macros/utils/cross_db_utils/surrogate_key.sql

```diff
@@ -0,0 +1,30 @@
+{#
+   Copyright 2021 dbt Labs, Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+#}
+
+
+{%- macro surrogate_key_backcompat(fields) -%}
+  {% set default_null_value = "" %}
+  {%- set field_sqls = [] -%}
+  {%- for field in fields -%}
+    {%- do field_sqls.append(
+        "coalesce(cast(" ~ field ~ " as " ~ type_string() ~ "), '" ~ default_null_value  ~"')"
+    ) -%}
+    {%- if not loop.last %}
+        {%- do field_sqls.append("'-'") -%}
+    {%- endif -%}
+  {%- endfor -%}
+  {{ hash(concat(field_sqls)) }}
```

Comment:
What are those `hash` and `concat` in the Jinja context?
I think those are macros of `dbt-utils` so maybe need to add package scoping.